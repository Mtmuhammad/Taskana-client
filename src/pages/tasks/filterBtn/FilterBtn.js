// Button to filter task results by status, completion, or importance
import React from 'react'

const FilterBtn = ({ onClick, title, icon, tasks }) => {
  return (
    <div className="col-12 col-lg-4">
      <div
        data-testid="filter-btn"
        id={title}
        onClick={onClick}
        className="task-filter"
      >
        <div data-testid="filter-icon" className="task-filter-icon">
          <i className={`d-none d-sm-block ${icon}`}></i>
        </div>
        <div className="d-flex align-items-center justify-content-between w-100">
          <h3 data-testid="filter-title" className="task-filter-title">
            {title}
          </h3>
          <span
            data-testid="filter-number"
            className={`task-filter-count badge badge-pill px-3 ${title}`}
          >
            {tasks && tasks.length}
          </span>
        </div>
      </div>
    </div>
  );
};

export default FilterBtn;
