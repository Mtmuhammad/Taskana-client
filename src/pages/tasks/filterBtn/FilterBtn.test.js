import React from "react";
import { render, screen } from "@testing-library/react";
import FilterBtn from "./FilterBtn";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<FilterBtn />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<FilterBtn />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("filter btn", () => {
  test("should render filter btn", () => {
    render(<FilterBtn />, { wrapper: BrowserRouter });
    const btn = screen.getByTestId("filter-btn");
    expect(btn).toBeInTheDocument();
    expect(btn).toHaveClass("task-filter");
  });
  test("should render filter icon", () => {
    render(<FilterBtn />, { wrapper: BrowserRouter });
    const icon = screen.getByTestId("filter-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("task-filter-icon");
  });
  test("should render filter title", () => {
    render(<FilterBtn />, { wrapper: BrowserRouter });
    const title = screen.getByTestId("filter-title");
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass("task-filter-title");
  });
  test("should render filter number", () => {
    render(<FilterBtn />, { wrapper: BrowserRouter });
    const number = screen.getByTestId("filter-number");
    expect(number).toBeInTheDocument();
    expect(number).toHaveClass("task-filter-count badge badge-pill px-3");
  });
});
