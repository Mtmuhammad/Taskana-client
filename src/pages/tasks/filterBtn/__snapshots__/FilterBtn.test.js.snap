// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12 col-lg-4"
  >
    <div
      class="task-filter"
      data-testid="filter-btn"
    >
      <div
        class="task-filter-icon"
        data-testid="filter-icon"
      >
        <i
          class="d-none d-sm-block undefined"
        />
      </div>
      <div
        class="d-flex align-items-center justify-content-between w-100"
      >
        <h3
          class="task-filter-title"
          data-testid="filter-title"
        />
        <span
          class="task-filter-count badge badge-pill px-3 undefined"
          data-testid="filter-number"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
